src/lib/storage/chunk_encoder.hpp

```diff
@@ -47,17 +52,39 @@ class ChunkEncoder {
   static void encode_chunk(const std::shared_ptr<Chunk>& chunk, const std::vector<DataType>& data_types,
                            const ChunkEncodingSpec& encoding_spec);
 
+  /**
+   * @brief Encodes a chunk using the same column-encoding spec
+   */
+  static void encode_chunk(const std::shared_ptr<Chunk>& chunk, const std::vector<DataType>& data_types,
+                           const ColumnEncodingSpec& column_encoding_spec = {});
```

Comment:
What are the semantics of passing an empty specification?

Reply:
The default constructor sets the encoding type to (deprecated) dictionary encoding at the moment.

Reply:
I am not sure if we want to consider one compression type to be the default. @Bouncner ?

Reply:
Currently, all tests assume dictionary encoding.

Reply:
I would vote for dict-encoding (probably byte-encoded) to be the default. That's what we teach and what students understand and that's what you get unless you actively decide to fiddle around with the encodings.

Or did I get your question wrong?

Reply:
Ok, I just wanted to make sure that we actively decide for this to be the default.